streaming/base/shared/prefix.py

```diff
@@ -184,29 +210,37 @@ def get_shm_prefix(streams_local: list[str],
     # Check my locals for overlap.
     _check_self(streams_local)
 
+    prefix_int = max([
+        _check_and_find_retrying(streams_local, streams_remote, shm_name=shm_name, retry=retry)
+        for shm_name in SHM_TO_CLEAN
+    ])
+
+    if dist.is_available() and dist.is_initialized():
+        dist.barrier()
+
     # First, the local leader registers the first available shm prefix, recording its locals.
     if world.is_local_leader:
-        prefix_int = _check_and_find_retrying(streams_local, streams_remote, retry)
         name = _get_path(prefix_int, LOCALS)
         data = _pack_locals(streams_local, prefix_int)
         shm = SharedMemory(name, True, len(data))
         shm.buf[:len(data)] = data
+        their_locals, their_prefix_int = _unpack_locals(bytes(shm.buf))
 
     if dist.is_available() and dist.is_initialized():
         dist.barrier()
 
     # Non-local leaders go next, searching for match.
     if not world.is_local_leader:
-        for prefix_int in _each_prefix_int():
-            name = _get_path(prefix_int, LOCALS)
-            try:
-                shm = SharedMemory(name, False)
-            except FileNotFoundError:
-                raise RuntimeError(f'Internal error: shared memory prefix was not registered by ' +
-                                   f'local leader. This may be because you specified ' +
-                                   f'different ``local`` parameters from different ranks.')
-            their_locals, their_prefix_int = _unpack_locals(bytes(shm.buf))
-            if streams_local == their_locals and prefix_int == their_prefix_int:
-                break
-
+        name = _get_path(prefix_int, LOCALS)
+        try:
+            shm = SharedMemory(name, False)
+        except FileNotFoundError:
+            raise RuntimeError(f'Internal error: shared memory prefix was not registered by ' +
+                               f'local leader. This may be because you specified ' +
+                               f'different ``local`` parameters from different ranks.')
+
+        their_locals, their_prefix_int = _unpack_locals(bytes(shm.buf))
+        if streams_local != their_locals or prefix_int != their_prefix_int:
```

Comment:
Should we add a debug log to print all these variables before we raise an exception?

Reply:
or maybe print these variables in the RuntimeError message.

Reply:
Added to err msg.